Fix bug Stocks Screener & Update finvizfinance package

minhhoang1023 opened the pull request:
# Description

- Fixes issue #1441: Finfizfinance package is outdated and not able to scrape the website. Update to latest version and refactor affected code.
- Fixes issue #1420: Add custom class to allow for space between argsparse arguments


<img width="797" alt="image" src="https://user-images.githubusercontent.com/40023817/157028353-1f739304-1032-41e4-b236-2b40dacee0cc.png">

- [ ] Summary of the change / bug fix.
- [ ] Link # issue, if applicable.
- [ ] Screenshot of the feature or the bug before/after fix, if applicable.
- [ ] Relevant motivation and context. 
- [ ] List any dependencies that are required for this change.


# How has this been tested?

* Please describe the tests that you ran to verify your changes. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.


# Checklist:

- [ ] Update [our Hugo documentation](https://gamestonkterminal.github.io/GamestonkTerminal/) following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/scripts).


# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [ ] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

jmaslek commented:
Just looking at the screen shot: Finviz title is misspelled, we can't see the tickers there and it still says that Market Cap isn't a valid sort column.

minhhoang1023 commented:
@jmaslek Nice spot on the misspellings!  

`Market Caps` should throw an error as only `Market Cap` is accepted. I intentionally checked for it as I have removed the choice option in argparse and replaced it with something else.

Ticker not clear is due to me maximizing the search sidebar on my 13' laptop 🤣 However, now several columns are not displaying full figures. Gonna add KMBTP magnitude to see if it's better.


<img width="1044" alt="image" src="https://user-images.githubusercontent.com/40023817/157041457-e40f8d12-8f00-4351-9cb7-6194add75373.png">

minhhoang1023 commented:
<img width="1027" alt="image" src="https://user-images.githubusercontent.com/40023817/157422538-259bc32c-4226-4655-a3ec-86f363cb35e4.png">

minhhoang1023 commented on gamestonk_terminal/stocks/screener/finviz_view.py:
```diff
@@ -163,15 +167,15 @@ def screener(
         df_screen = df_screen.dropna(axis="columns", how="all")
 
         if sort:
-            if " ".join(sort) in d_cols_to_sort[data_type]:
```
@DidierRLopes Here I updated that only one column can be selected for sorting. More than one column we'd lose the auto-complete functionality. And it would require more dev time for it to be compatible with white space in column names.

DidierRLopes reviewed (changes requested):
<img width="1541" alt="Screenshot 2022-03-10 at 10 23 08" src="https://user-images.githubusercontent.com/25267873/157642110-01e8c7c9-93ff-4fd4-8d57-69bd0d7cc266.png">

It says that "Market Cap" is an option but then doesn't allow for it. My suggestion is to create a dictionary where keys correspond to 1 word per column and description is the name the column. This way we can use the dict.keys() on the auto-complete, and the then when sorting we can use dict[ns_parser.sort] value. Does that make sense?

<img width="1544" alt="Screenshot 2022-03-10 at 10 25 29" src="https://user-images.githubusercontent.com/25267873/157642469-8484ee9e-9391-4f50-b3e2-61a9f023cc36.png">

The sorting of the data frame shouldn't be necessary to output something. 

Also, can we mute the 
```
[Info] loading page ##############################################################################################################################################################
```
It's very noisy. If there was just one would be fine, but multiple makes it look bad.

You should be able to do this through an argument, but if not you can use:`with suppress_stdout():`

minhhoang1023 commented:
technical:
<img width="1074" alt="image" src="https://user-images.githubusercontent.com/40023817/157735663-b3b96487-dde4-49a4-80ac-4b284cebd2f9.png">

performance:
<img width="1069" alt="image" src="https://user-images.githubusercontent.com/40023817/157735755-70615cf5-e6a7-4cfd-a6e7-342612f5240d.png">

ownership:
<img width="1076" alt="image" src="https://user-images.githubusercontent.com/40023817/157735831-fef7d436-b374-4233-b11e-ebf046878b72.png">

valuation:
<img width="1076" alt="image" src="https://user-images.githubusercontent.com/40023817/157735883-67851e69-0b0c-4510-b6b8-5bbf6432ff40.png">
